Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -592,7 +592,7 @@ uint64_t Connection::sendWithAsyncReply(T&& message, C&& completionHandler, uint
             T::callReply(*decoder, WTFMove(completionHandler));
         else
             T::cancelReply(WTFMove(completionHandler));
-    }, CompletionHandlerCallThread::MainThread));
+    }));
```

Comment:
I think this is the behavior currently:
- sendWithAsyncReply on main run loop
  - async reply delivered on main run loop
- sendWithAsyncReply other thread, work queue, run loop
 - async reply delivered on main run loop
 
 
I think this is the behavior after your change:
- sendWithAsyncReply on main run loop
  - async reply delivered on main run loop
- sendWithAsyncReply other thread, work queue, run loop
 - assert fails when the reply is delivered
 
 
 I'm in process of implementing the run loop binding of the connection. I'll add the test cases for this, too, when implementing it.